pkg/volume/emptydir/empty_dir.go

```diff
@@ -110,14 +110,8 @@ func (plugin *emptyDirPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod) (volume
 }
 
 func calculateEmptyDirMemorySize(nodeAllocatableMemory *resource.Quantity, spec *volume.Spec, pod *v1.Pod) *resource.Quantity {
-	// if feature is disabled, continue the default behavior of linux host default
-	sizeLimit := &resource.Quantity{}
-	if !utilfeature.DefaultFeatureGate.Enabled(features.SizeMemoryBackedVolumes) {
```

Comment:
This all needs to stay because of version-compat.  Right @jpbetz  ?

https://github.com/kubernetes/community/issues/8082

Reply:
There seems to be some confusion here so I'll wait for consensus. It makes sense to remove this to me as this is effectively dead code. (GA feature gates will never go to false). I think we should not remove the true part of the feature gate but I will wait on consensus.

Reply:
Yes, please keep the feature gate code. Yes it used to be dead code but with the introduction of [Compatibility Versions](https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/4330-compatibility-versions/README.md), kubernetes is able to emulate a previous version that could have the feature gate disabled. We are in the process of updating the documentation for all this, thank you for catching this Tim!

Reply:
Okay. I will add this back. So when we remove the feature gate in n+3, we can remove the dead code.

Reply:
removed.

Reply:
Yeah, instead if it being "set GA and delete gated code; wait +2; remove gate" it's now "set GA; wait +3; delete gated code and remove gate"

Reply:
> We are in the process of updating the documentation for all this

I am a bit confused how compatibility mode can be enabled and what the "LockToDefault" would mean. Documentation would be awesome to have. I am mostly interested on how we will guarantee functionality without the feature gate and what testing is run (or needs to run) to ensure functionality.

Reply:
Thanks for the feedback and this is definitely something that we will clarify in the documentation. 

Per the [Compatibility Version KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/4330-compatibility-versions/README.md#--emulation-version), a k8s cluster at a particular binary version can emulate a previous version up to three releases before (min version starts at 1.31). Features/APIs/storage-versions/validation will emulate the behavior at the target emulation version. So a 1.32 binary version cluster emulating 1.31 will have `SizeMemoryBackedVolumes` in beta and disableable. LockToDefault is only applied to emulation version 1.32+ (hence why feature promotions are all versioned now).

Reply:
Wait, KEP explicitly says: "Adding --emulation-version to kubelet is out of scope for this enhancement."

I was always under impression that for nodes we are saying just to install a previous kubelet version. Has this changed?

Reply:
xref to another KEP to update on the decision here: https://github.com/kubernetes/kubernetes/pull/128046#discussion_r1799759192

Reply:
> I was always under impression that for nodes we are saying just to install a previous kubelet version. Has this changed?

Correct, kubelet is unaffected by compatibility versions.